### ISSUE TITLE
Capture the screen and crop it when taking screenshots on Windows

### ISSUE THREAD

**Issue by AndreyBelym** [DevExpress/testcafe-browser-natives#75]
**TL;DR**: We can use `SetWindowPos` to bring the specified window on foreground. It works instantly and reliable at least on Windows 10. I propose at least to use it to bring MS Edge to foreground instead of our hack, after testing on other Windows versions. Or get rid of `PrintWindow` at all and use screen grabbing for all browsers.

###### Explanation  
 
The problem with taking a screenshot of the particular window on MS Windows is that `PrintWindow` function, that we are currently using, is very unreliable, and often produces black images because it doesn't work with hardware accelerated windows.

Looks like the only reliable way to capture the specified window is grabbing the entire screen image, and then crop it to the window size. Here the trouble is that the specified window must be foreground, on top of window z-ordered stack. Note that [SetForegroundWindow](https://msdn.microsoft.com/ru-ru/library/windows/desktop/ms633539(v=vs.85).aspx) have a very restrictive rules when it may be used (most important is that the process using it must be foreground process, or process with active user input), so we can use it.

Currently we are using similar mechanism when dealing with MS Edge browser, because `PrintWindow` does not work with new Universal Windows Applications. We bring it to foreground, using a hack, then capture the screen and crop it to Edge window size and position. The hack is minimizing the window and then restoring it, which is slow due to window animations and works unreliable with Chrome and Firefox, if multiple windows of these browsers are opened.

Now I have found [another way](http://owl-black.net/479-how-to-change-z-order-of-a-window) to bring a window to the top with `SetWindowPos`, . I remember that I tried to use it when I wrote the module a year ago, but it didn't work for me at that time. I had Windows 8.1, but now I've used it on Windows 10, and it works.

**Comment by AlexanderMoskovkin** [comment]
I think we can try this. But it should work on win7, 8 and 10

**Comment by AndreyBelym** [comment]
Turns out that there is a new feature in Windows 10, that creates obstacles for creating screenshots this way: [Virtual Desktops](https://blogs.windows.com/windowsexperience/2015/04/16/virtual-desktops-in-windows-10-the-power-of-windowsmultiplied/). When a window is hidden in a non-active virtual desktop, it will be non-trivial to place it on the active desktop, and it will require hooks injection to the browser process. So we need to check if the window is on the active virtual desktop, and if it isn't, use the the old way with the `PrintWindow` function.
